src/lava/magma/runtime/runtime_service.py

```diff
@@ -182,12 +162,15 @@ def run(self):
         step. The loop ends when receiving the STOP command from the runtime."""
         selector = CspSelector()
         phase = LoihiPyRuntimeService.Phase.HOST
+
+        channel_actions = [(self.runtime_to_service, lambda: 'cmd')]
+
         while True:
             # Probe if there is a new command from the runtime
-            cmd = selector.select((self.runtime_to_service_cmd, lambda: True),
-                                  (self.runtime_to_service_req, lambda: False))
-            if cmd:
-                command = self.runtime_to_service_cmd.recv()
+            action = selector.select(*channel_actions)
```

Comment:
Since channel_actions only has one channel, the select could be simplified to just a receive directly on the channel, unless more channels are anticipated to be added later.

Reply:
Yes, I think this is first part of the refactoring. We will add polling on more channels as we implement the pause behaviour.